cmd/prometheus/main.go

```diff
@@ -152,6 +155,7 @@ type flagConfig struct {
 	enableNewSDManager         bool
 	enablePerStepStats         bool
 	enableAutoGOMAXPROCS       bool
+	enableKubeSecretProvider   bool
```

Comment:
That's true, and this is intended. The idea is that when the Prometheus upstream version is eventually merged, it will not be compatible with this one. We may want to keep both implementations until we update to the upstream version within prometheus-engine.